synapse/handlers/profile.py

```diff
@@ -166,10 +171,16 @@ def set_displayname(self, target_user, requester, new_displayname, by_admin=Fals
         )
 
         if self.hs.config.user_directory_search_all_users:
-            profile = yield self.store.get_profileinfo(target_user.localpart)
```

Comment:
I'm not entirely sure I'm happy that we end up with a UserDirectoryHandler which we then carefully don't use. Could we have some sort of stub handler which just provides a `notify_profile_changed` and is implemented by different classes on the workers than on the master?